### PR TITLE
Add a filter to allow the `feature` index of each module to be filtered

### PR DESCRIPTION
**Pull request by simonwheatley**
Addresses #1939

### CODE REVIEW

**Comment by simonwheatley** [comment]
Example code to hook this filter

```php
/**
 * Filter hook fired on the jetpack_recommended_modules filter. Here we ensure that
 * Beautiful Math (latex), Sharing (sharedaddy), and Carousel (carousel) are all 
 * included in the recommended modules.
 */
function sw_filter_jetpack_module_feature( $module_featured, $module, $module_data ) {
	$to_feature = array( 'latex', 'sharedaddy', 'carousel' );
	if ( in_array( $module, $to_feature ) ) {
		$module_featured = array( 'Recommended' );
	}
	return $module_featured;
}
add_filter( 'jetpack_module_feature', 'sw_filter_jetpack_module_feature', 10, 3 );
```

**Comment by simonwheatley** [comment]
I am uncertain I've described the possible array values correctly in the code doc, so can someone please verify these?

File: class.jetpack.php [hunk 1]
```diff
@@ -1801,6 +1801,23 @@ public static function get_module( $module ) {
...
+		 * Jumpstart, and the default "Other" listing.
+		 *
+		 * @since 3.5
+		 *
+		 * @param array   $mod['feature'] The areas to feature this module:
+		 *     'Jumpstart' adds to the "Jumpstart" option to activate many modules at once
+		 *     'Recommended' shows on the main Jetpack admin screen
+		 *     'Other' should be the default if no other value is in the array
+		 * @param string  $module The slug of the module, e.g. sharedaddy
+		 * @param type    $mod All the currently assembled module data
```
Comment (dereksmart) [inline class.jetpack.php]: This param type is `array`

**Comment by dereksmart** [comment]
Made a small note about the docblock, but otherwise LGTM.  

Works well, except right now we're limiting the amount of recommended modules shown to 9 (via css).  It's a quick fix. @samhotchkiss or @simonwheatley would you like me to create a separate PR with that change or just merge into this branch? 

Might I also suggest adding to your example code to include an `else` statement that would illustrate how to only show the modules filtered in, rather than adding them on? 

```php
if ( in_array( $module, $to_feature ) ) {
    $module_featured = array( 'Recommended' );
 } else {
    $module_featured = array( 'Other' );
}
```

**Comment by dereksmart** [comment]
I've pushed the change to this branch, so we're not limiting the number of featured modules.  @simonwheatley I hope you don't mind!

**Comment by samhotchkiss** [comment]
Looks good to me, aside from all the grunt noise...

**Comment by rase-** [comment]
LGTM as well, but yeah quite a bit of grunt noise. Likely needs a rebase for that reason as well.

**Comment by dereksmart** [comment]
Rebased and will be ready to merge once travis gives it :+1:

**Comment by simonwheatley** [comment]
@dereksmart Missed your notes in the busy Jetpack stream, all seems OK to me :+1:

**Pull request merged**